core/src/main/java/org/apache/accumulo/core/data/Mutation.java

```diff
@@ -772,6 +772,236 @@ public void putDelete(byte[] columnFamily, byte[] columnQualifier, ColumnVisibil
     put(columnFamily, columnQualifier, columnVisibility.getExpression(), true, timestamp, true, EMPTY_BYTES);
   }
 
+  /**
+   * Fluent API for Mutation
+   *
+   * <p>
+   * Each step of the Mutation is represented by an interface. Inheritance order ensures that chained methods follow prescribed order of:
+   *
+   * <p>
+   * row, family, qualifier, visibility, timestamp
+   *
+   * <p>
+   * set and delete methods end the chain and finalize the mutation
+   */
+  public interface RowStep extends FamilyStep {
```

Comment:
What happens if someone creates a mutation with a specific row and then uses `row()` with a different one?

```java
Mutation m = new Mutation("row1");
m.at().row("row2").family("fam").set("val");
```

Reply:
There should not be a row step.  All changes to a mutation are for a single row.

Reply:
I wasn't sure about this one so I didn't include any implementation for it. Thanks for the clarification! I've removed the row methods and interface in my most recent commit.